### PR TITLE
python3Packages.openllm: fix `optional-dependencies` attribute

### PR DESCRIPTION
**Pull request by trofi**
Without the change `python3Packages.openllm.optional-dependencies.agents` always fails to evaluate as:

    $ nix build --no-link -f. python3Packages.openllm.optional-dependencies.agents
    error:
       error: attribute 'agents' missing
           93|       transformers
           94|     ] ++ transformers.optional-dependencies.agents;
             |          ^
           95|     awq = [

As `transformers` never exposed `agents` let's just drop the reference.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/openllm/default.nix [hunk 1]
```diff
@@ -91,7 +91,7 @@ buildPythonPackage rec {
       # diffusers
       soundfile
       transformers
-    ] ++ transformers.optional-dependencies.agents;
+    ];
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/openllm/default.nix:94]: Did you look into if we could add that field instead?

Comment (trofi) [reply pkgs/development/python-modules/openllm/default.nix:94]: Nope. I don't know how those attributes work or how to test them properly. I can only build them as is :)

It probably should provide something at  https://github.com/huggingface/transformers/blob/3cefac1d974db5e2825a0cb2b842883a628be7a0/setup.py#L412 . But that would be new functionality which a familiar maintainer should be able to test.

Comment (SuperSandro2000) [reply pkgs/development/python-modules/openllm/default.nix:94]: > Nope. I don't know how those attributes work or how to test them properly.

optional-dependencies is just a list of python packages which can be optionally enabled. No magic and there is nothing really to test.



> a familiar maintainer should be able to test.

It's a python package. Likely you and me already know more than most person by just looking at this package. :)
---

We probably just need to add the following to transformers to make this work:

```nix
passthru = {
  optional-dependencies.agents = [ diffusers accelerate datasets torch sentencepiece opencv-python pillow ];
};
```
and maybe add a few new inputs.

Comment (trofi) [reply pkgs/development/python-modules/openllm/default.nix:94]: Sounds good. I'll give it a go. Converting to draft for now. If it works we might need to revert the other two PRs:
- https://github.com/NixOS/nixpkgs/pull/277788
- https://github.com/NixOS/nixpkgs/pull/277777

Comment (trofi) [reply pkgs/development/python-modules/openllm/default.nix]: Exposed `optional-dependencies.agents` to `python3Packages.transformers`.

**Pull request merged**